ompi/mca/mtl/ofi/mtl_ofi_component.c

```diff
@@ -411,6 +429,76 @@ ompi_mtl_ofi_component_init(bool enable_progress_threads,
         goto error;
     }
 
+    /* Check if FI_REMOTE_CQ_DATA is supported, if so send the source rank there
+     * FI_DIRECTED_RECV is also needed so receives can discrimate the source
+     */
+    prov_tmp_name = strdup(prov->fabric_attr->prov_name);
+    if(!prov_tmp_name){
```

Comment:
in OMPI, we explicitly check against the NULL.

I also don't understand why you're swapping provider names here; a comment would go a long way.

Reply:
Not swapping providers names, jut making a copy to avoid issues with fi_freeinfo().  Will add a comment.